playback/tape_cassettes/s3/s3_tape_cassette.py

```diff
@@ -267,24 +296,20 @@ def iter_recording_ids(self, category, start_date=None, end_date=None, metadata=
         :type metadata: dict
         :param limit: Optional limit on number of ids to fetch
         :type limit: int
+        :param random_results: True to return result in random order
+        :type random_results: bool
         :return: Iterator of keys matching the given parameters
         :rtype: collections.Iterator[basestring]
         """
         content_filter = self._create_content_filter_func(metadata) if metadata else None
 
-        # to improve performance when looking for recordings in s3, the date is added to the folder
-        # and when a start date is given we can look for specific folders until today (or end_time)
-        if start_date:
-            end_date = end_date or datetime.utcnow()
-            days = [(start_date + timedelta(days=i)) for i in range((end_date - start_date).days + 1)]
-            id_prefixes = ['{}/{}/'.format(category, day.strftime(self.DAY_FORMAT)) for day in days]
-        else:
-            id_prefixes = ['{}/'.format(category)]
+        id_prefixes = self._get_id_prefixes(category, start_date, end_date)
 
-        id_prefix_iterators = self.create_id_prefix_iterators(id_prefixes, start_date, end_date, content_filter, limit)
+        id_prefix_iterators = self.create_id_prefix_iterators(id_prefixes, start_date, end_date, content_filter, limit,
+                                                              random_results)
         count = 0
         while count != limit and id_prefix_iterators:
-            random_day_iterator = random.choice(id_prefix_iterators)
+            random_day_iterator = random.choice(id_prefix_iterators) if random_results else id_prefix_iterators[0]
```

Comment:
Does this mean when Random is off you always extract from the first day iterator until limit is reached?
That would make local runs (which don't have random enabled) sample only last day.

Perhaps we should do round robin and every time pick one from another iterator but going over next iterator day each time?

Reply:
Ok it is sound reasonable, I will change it